src/components/SelectionListRadio/BaseSelectionListRadio.js

```diff
@@ -0,0 +1,378 @@
+
+    const [focusedIndex, setFocusedIndex] = useState(() => {
+        const defaultIndex = 0;
+
+        const indexOfInitiallyFocusedOption = _.findIndex(flattenedSections.allOptions, (option) => option.keyForList === props.initiallyFocusedOptionKey);
+
+        if (indexOfInitiallyFocusedOption >= 0) {
+            return indexOfInitiallyFocusedOption;
+        }
+
+        return defaultIndex;
+    });
+
+    /**
+     * Scrolls to the focused index within the SectionList
+     *
+     * @param {Number} index
+     */
+    const scrollToIndex = useCallback(
+        (index) => {
+            const item = flattenedSections.allOptions[index];
+
+            if (!listRef.current || !item) {
+                return;
+            }
+
+            const itemIndex = item.index;
+            const sectionIndex = item.sectionIndex;
+
+            // Note: react-native's SectionList automatically strips out any empty sections.
+            // So we need to reduce the sectionIndex to remove any empty sections in front of the one we're trying to scroll to.
+            // Otherwise, it will cause an index-out-of-bounds error and crash the app.
+            let adjustedSectionIndex = sectionIndex;
+            for (let i = 0; i < sectionIndex; i++) {
+                if (_.isEmpty(lodashGet(props.sections, `[${i}].data`))) {
+                    adjustedSectionIndex--;
+                }
+            }
+
+            listRef.current.scrollToLocation({sectionIndex: adjustedSectionIndex, itemIndex, animated: true});
+        },
+        [flattenedSections.allOptions, props.sections],
+    );
+
+    /**
+     * This function is used to compute the layout of any given item in our list.
+     * We need to implement it so that we can programmatically scroll to items outside the virtual render window of the SectionList.
+     *
+     * @param {Array} data - This is the same as the data we pass into the component
+     * @param {Number} flatDataArrayIndex - This index is provided by React Native, and refers to a flat array with data from all the sections. This flat array has some quirks:
+     *
+     *     1. It ALWAYS includes a list header and a list footer, even if we don't provide/render those.
+     *     2. Each section includes a header, even if we don't provide/render one.
+     *
+     *     For example, given a list with two sections, two items in each section, no header, no footer, and no section headers, the flat array might look something like this:
+     *
+     *     [{header}, {sectionHeader}, {item}, {item}, {sectionHeader}, {item}, {item}, {footer}]
+     *
+     * @returns {Object}
+     */
+    const getItemLayout = (data, flatDataArrayIndex) => {
+        const targetItem = flattenedSections.itemLayouts[flatDataArrayIndex];
+
+        return {
+            length: targetItem.length,
+            offset: targetItem.offset,
+            index: flatDataArrayIndex,
+        };
+    };
+
+    const renderItem = ({item, index, section}) => {
```

Comment:
The same as above, maybe we could introduce tests to measure render count?

Reply:
Great idea - I created a new component `RadioListItem` in a new file, that should already prevent re-renders. Also created a perf test `SelectionListRadio.perf-test.js`